pywinauto/keyboard.py

```diff
@@ -630,6 +638,20 @@ def SendKeys(keys,
         """Parse the keys and type them"""
         keys = parse_keys(keys, with_spaces, with_tabs, with_newlines)
 
-        for k in keys:
-            k.run()
-            time.sleep(pause)
+        _run(keys, pause)
+
+    def _key_up_or_down(key, up, down):
+        keys = parse_keys(key)
+        if isinstance(keys[0].key, six.string_types):
+            keys[0] = EscapedKeyAction(keys[0].key)
+        keys[0].down = down
+        keys[0].up = up

```

Comment:
Since we need marking keys as "up" or "down" in `SendKeys` method, this logic could be moved to `parse_keys`.

Reply:
@vasily-v-ryabov I moved the entire logic to handle the combination of keys you mentioned above apart from the existing ones to "parse_keys" function . Please let me know your comments.

```diff
diff --git "a/C:\\Users\\kbad\\AppData\\Local\\Temp\\TortoiseGit\\keyboard-ed91b70.002.py" "b/C:\\MyData\\NonWork\\open-source\\pywinauto\\pywinauto\\keyboard.py"
index 5116f71..9b437b8 100644
--- "a/C:\\Users\\kbad\\AppData\\Local\\Temp\\TortoiseGit\\keyboard-ed91b70.002.py"
+++ "b/C:\\MyData\\pywinauto\\pywinauto\\keyboard.py"
@@ -496,7 +496,7 @@ else:
                 try:
                     pause_time = float(count)
                 except ValueError:
-                    raise KeySequenceError('invalid pause time %s' % count)
+                    raise KeySequenceError('invalid pause time %s'% count)
                 code_keys.append(PauseAction(pause_time))
 
             else:
@@ -524,7 +524,6 @@ else:
 
         return code_keys
 
-
     def parse_keys(string,
                    with_spaces=False,
                    with_tabs=False,
@@ -534,6 +533,8 @@ else:
         keys = []
         if not modifiers:
             modifiers = []
+
+        should_escape_next_keys = False
         index = 0
         while index < len(string):
 
@@ -563,13 +564,27 @@ else:
             # Escape or named key
             elif c == "{":
                 # We start searching from index + 1 to account for the case {}}
-                end_pos = string.find("}", index + 1)
+                end_pos = string.find("}", index+1)
                 if end_pos == -1:
                     raise KeySequenceError('`}` not found')
 
                 code = string[index:end_pos]
                 index = end_pos + 1
-                keys.extend(handle_code(code))
+                key_events = [' up', ' down', ' UP', ' DOWN']
+                current_key_event = None
+                if any(key_event in code for key_event in key_events):
+                    code, current_key_event = code.split(' ')
+                    should_escape_next_keys = True
+                current_keys = handle_code(code)
+                if current_key_event is not None:
+                    if isinstance(current_keys[0].key, six.string_types):
+                        current_keys[0] = EscapedKeyAction(current_keys[0].key)
+
+                    if current_key_event.strip() == 'up':
+                        current_keys[0].down = False
+                    else:
+                        current_keys[0].up = False
+                keys.extend(current_keys)
 
             # unmatched ")"
             elif c == ')':
@@ -596,7 +611,7 @@ else:
                 # if ord(c) in CODE_NAMES:
                 #    keys.append(VirtualKeyAction(ord(c)))
 
-                elif modifiers:
+                elif modifiers or should_escape_next_keys:
                     keys.append(EscapedKeyAction(c))
 
                 else:
@@ -624,10 +639,6 @@ else:
         """Return the high byte of the value"""
         return (val & 0xff00) >> 8
 
-    def _run(keys, pause):
-        for k in keys:
-            k.run()
-            time.sleep(pause)
 
     def SendKeys(keys,
                  pause=0.05,
@@ -638,20 +649,6 @@ else:
         """Parse the keys and type them"""
         keys = parse_keys(keys, with_spaces, with_tabs, with_newlines)
 
-        _run(keys, pause)
-
-    def _key_up_or_down(key, up, down):
-        keys = parse_keys(key)
-        if isinstance(keys[0].key, six.string_types):
-            keys[0] = EscapedKeyAction(keys[0].key)
-        keys[0].down = down
-        keys[0].up = up
-        _run(keys, pause=0.01)
-
-    def key_up(key):
-        """Parses the given key and releases it"""
-        _key_up_or_down(key, up=True, down=False)
-
-    def key_down(key):
-        """Parses the given key and presses and holds it"""
-        _key_up_or_down(key, up=False, down=True)
+        for k in keys:
+            k.run()
+            time.sleep(pause)


```

Reply:
It's easier to leave comment for normal commit than for a diff. The approach seems OK to me. Just need a minor improvement: using str method `.lower()` for checking only `"up"` and `"down"` substrings inclusion.

Reply:
FYI: commit to the same branch will update the PR automatically.

Reply:
@vasily-v-ryabov Sorry about it. I didn't want to start the automatic build process until I know it's correct by design. Anyway, I pushed the changes for your review. Thanks!